widget/embedded/src/store/quote.ts

```diff
@@ -289,3 +296,55 @@ export const useQuoteStore = createSelectors(
     }))
   )
 );
+
+useQuoteStore.subscribe((selectedState, previousSelectedState) => {
+  if (
+    selectedState.fromBlockchain !== previousSelectedState.fromBlockchain ||
+    selectedState.fromToken !== previousSelectedState.fromToken ||
+    selectedState.toBlockchain !== previousSelectedState.toBlockchain ||
+    selectedState.toToken !== previousSelectedState.toToken ||
+    selectedState.inputAmount !== previousSelectedState.inputAmount
+  ) {
+    // useEffect hook can not be used in Zustand subscribe
+    eventEmitter.emit(MainEvents.QuoteEvent, {
+      type: QuoteEventTypes.QUOTE_INPUT_UPDATE,
+      payload: {
+        fromBlockchain: selectedState.fromBlockchain?.name,
+        toBlockchain: selectedState.toBlockchain?.name,
+        fromToken: selectedState.fromToken
+          ? {
+              symbol: selectedState.fromToken?.symbol,
+              name: selectedState.fromToken?.name,
+              address: selectedState.fromToken?.address,
+            }
+          : undefined,
+        toToken: selectedState.toToken
+          ? {
+              symbol: selectedState.toToken?.symbol,
+              name: selectedState.toToken?.name,
+              address: selectedState.toToken?.address,
+            }
+          : undefined,
+        requestAmount: selectedState.inputAmount,
+      },
+    });
+  }
+
+  if (
+    selectedState.selectedQuote?.requestId !==
+    previousSelectedState.selectedQuote?.requestId
+  ) {
+    eventEmitter.emit(MainEvents.QuoteEvent, {
+      type: QuoteEventTypes.QUOTE_UPDATE,
+      payload: selectedState.selectedQuote
+        ? {
+            requestAmount: selectedState.selectedQuote?.requestAmount,
+            swaps: selectedState.selectedQuote?.swaps,
+            outputAmount: selectedState.selectedQuote?.outputAmount,
+            resultType: selectedState.selectedQuote?.resultType,
+            tags: selectedState.selectedQuote?.tags,
```

Comment:
The `selectedQuote` has value here. you can remove `?`

Reply:
Tnx fixed